### ISSUE TITLE
mocha-typescript-watch regular expressions pass-through

### ISSUE THREAD

**Issue by truhlikfredy** [pana-cc/mocha-typescript#19]
Mocha allows regular expressions to match test cases and you can decide which tests to skip. Would be nice if the mocha-typescript-watch would allow to pass-through these parameters back to mocha.

```
  -g, --grep <pattern>                    only run tests matching <pattern>
  -f, --fgrep <string>                    only run tests containing <string>
```

**Comment by pana-cc** [comment]
You can register any arguments you would like to pass down to mocha here:
https://github.com/PanayotCankov/mocha-typescript/blob/master/bin/watch.ts#L37
And pass them to the mocha process here:
https://github.com/PanayotCankov/mocha-typescript/blob/master/bin/watch.ts#L71

Or alternatively we can think of a way to provide arbitrary arguments to mocha using something like `mocha-typescript-watch -- <everything-passes-down-to-mocha>`, but I will have to research what is supported in yargs.

**Comment by truhlikfredy** [comment]
You mean with something like  -m ./node_modules/mocha/bin/_mocha -g expression ?

**Comment by pana-cc** [comment]
@truhlikfredy, exactly, :)

This has been resolved within the 1.0.19 patch of the modules.
You can now execute:
``` bash
$ ./node_modules/.bin/mocha-typescript-watch -g <regex pattern>
```

Or should you have the package.json script:
``` JavaScript
  "scripts": {
    "test": "tsc && mocha",
    "watch": "mocha-typescript-watch"
  },
```
You can pass the -f or -g flags through `npm run` like:
``` bash
$ npm run watch -- -g <regex pattern>
```

As a side note I never use these flags, usually once I start the watcher, when I focus on a suite or test I add the `@test @only test() {}` decorator or change from `describe()` or `it()` to `describe.only()` or `it.only()`, and after a save, the watcher will compile and execute only the marked tests. I find it much more straight forward than having to copy the name to the command line and re-run the mocha script.

On the other hand the grep and fgrep are useful in case you want to run groups of tests scattered in various suites, so here it is :)
